platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -383,6 +390,18 @@ private void logout() {
 
         SecurityTokenHolder tokenHolder = ((SecurityTokenHolder) session.getAttribute(
                 SecurityConstants.SAML_ASSERTION));
+
+        for (SecurityToken token : tokenHolder.getRealmTokenMap()
+                .values()) {
+            try {
+                Subject subject = securityManager.getSubject(token);
```

Comment:
Using the securityManager to do this is kind of heavyweight. It'd be much faster to use SecurityAssertionImpl to rip apart the SecurityToken for you.